Source/Core/MapboxApi.js

```diff
@@ -1,54 +1,47 @@
-import Credit from "./Credit.js";
 import defined from "./defined.js";
+import deprecationWarning from "./deprecationWarning.js";
+
+var defaultAccessToken;
 
 /**
  * @namespace MapboxApi
  */
 var MapboxApi = {};
 
-/**
- * The default Mapbox API access token to use if one is not provided to the
- * constructor of an object that uses the Mapbox API.  If this property is undefined,
- * Cesium's default access token is used, which is only suitable for use early in development.
- * Please supply your own access token as soon as possible and prior to deployment.
- * Visit {@link https://www.mapbox.com/help/create-api-access-token/} for details.
- * When Cesium's default access token is used, a message is printed to the console the first
- * time the Mapbox API is used.
- *
- * @type {String}
- */
-MapboxApi.defaultAccessToken = undefined;
-
-var printedMapboxWarning = false;
-var errorCredit;
-var errorString =
-  "<b>This application is using Cesium's default Mapbox access token.  Please create a new access token for the application as soon as possible and prior to deployment by visiting <a href=https://www.mapbox.com/account/apps/>https://www.mapbox.com/account/apps/</a>, and provide your token to Cesium by setting the Cesium.MapboxApi.defaultAccessToken property before constructing the CesiumWidget or any other object that uses the Mapbox API.</b>";
+Object.defineProperties(MapboxApi, {
+  /**
+   * The default Mapbox API access token to use if one is not provided to the
+   * constructor of an object that uses the Mapbox API.  If this property is undefined,
+   * Cesium's default access token is used, which is only suitable for use early in development.
+   * Please supply your own access token as soon as possible and prior to deployment.
+   * Visit {@link https://www.mapbox.com/help/create-api-access-token/} for details.
+   * When Cesium's default access token is used, a message is printed to the console the first
+   * time the Mapbox API is used.
+   *
+   * @type {String}
+   * @memberof MapboxApi
+   * @deprecated
+   */
+  defaultAccessToken: {
+    set: function (value) {
+      defaultAccessToken = value;
+      deprecationWarning(
+        "mapbox-token",
```

Comment:
I would not make this change.  Instead, just leave `MapboxApi.defaultAccessToken = undefined;` and remove the token from `getAccessToken`.  It's convenient to set this once instead of having to pass the token to each `MapboxImageryProvider` you create.  In fact, that's exactly what the `errorString` for using the token we provided recommended.

Reply:
Disagree, it's an entire class we would be maintaining for no real reason.  It's absolutely trivial for them to just define a default access token in their own application code.